h2o-algos/src/test/java/hex/deeplearning/DeepLearningAutoEncoderCategoricalTest.java

```diff
@@ -84,16 +85,43 @@ public void run() {
 
     Assert.assertEquals(l2vec.mean(), mymodel.mse(), 1e-8*mymodel.mse());
 
+    // Create reconstruction
+    Log.info("Creating full reconstruction.");
+    final Frame recon_train = mymodel.score(train);
+    Assert.assertTrue(mymodel.testJavaScoring(train,recon_train,1e-5));
+
+    Frame df1 = mymodel.scoreDeepFeatures(train, 0);
+    Assert.assertTrue(df1.numCols() == 10);
+    Assert.assertTrue(df1.numRows() == train.numRows());
+    df1.delete();
+
+    Frame df2 = mymodel.scoreDeepFeatures(train, 1);
+    Assert.assertTrue(df2.numCols() == 5);
+    Assert.assertTrue(df2.numRows() == train.numRows());
+    df2.delete();
+
+    Frame df3 = mymodel.scoreDeepFeatures(train, 2);
+    Assert.assertTrue(df3.numCols() == 3);
+    Assert.assertTrue(df3.numRows() == train.numRows());
+    df3.delete();
+
+    // check if reconstruction error is the same from model and mojo model too. Testcase for PUBDEV-6030.
     try {
       DeeplearningMojoModel mojoModel = (DeeplearningMojoModel) mymodel.toMojo();
       EasyPredictModelWrapper model = new EasyPredictModelWrapper(mojoModel);
       AutoEncoderModelPrediction tmpPrediction;
       RowData tmpRow;
+      BufferedString bStr;
       double calcNormMse = 0;
       for (int r = 0; r < train.numRows(); r++) {
         tmpRow = new RowData();
+        bStr = new BufferedString();
```

Comment:
It doesn't matter in test but BufferedString is meant to be reused, it shouldn't be created inside of a loop.

Reply:
Ok.